pandas/tests/arithmetic/test_datetime64.py

```diff
@@ -2161,7 +2161,9 @@ def test_dti_isub_tdi(self, tz_naive_fixture):
         [
             datetime(2011, 1, 1),
             DatetimeIndex(["2011-01-01", "2011-01-02"]),
-            DatetimeIndex(["2011-01-01", "2011-01-02"]).tz_localize("US/Eastern"),
+            DatetimeIndex(["2011-01-01", "2011-01-02"]).__getattribute__("tz_localize")(
```

Comment:
I think the `PandasDelegate` class is responsible for adding `tz_localize` to the DTI. I don't know how to handle that type of paradigm with mypy so might need some research, but we don't want to change this

Reply:
Yeah. I think there's a valid argument to be made for saying mypy's not equipped to handle this properly - would a simple `# type: ignore` be the way to go? I guess we could also try declaring an intermediate variable. My local environment has something funky going on, local tests aren't working the same as the checks so I'll have to run the final tests against Travis. I'll keep working!

Reply:
now that out min version is 3.6 could just declare the attribute in the appropriate base class.